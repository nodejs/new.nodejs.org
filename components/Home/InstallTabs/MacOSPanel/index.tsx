import { FormattedMessage } from 'react-intl';
import LocalizedLink from '../../../LocalizedLink';
import { ShellBox } from '../../../Common';
import styles from '../index.module.scss';
import type { FC } from 'react';

type Props = {
  nvmVersion: string;
};

export const PureMacOSPanel: FC<Props> = ({ nvmVersion }) => {
  const nvmInstallScriptUrl = `https://raw.githubusercontent.com/nvm-sh/nvm/${nvmVersion}/install.sh`;

  return (
    <div>
      <ShellBox textToCopy={`curl -o- ${nvmInstallScriptUrl} | bash`}>
        <span className={styles.installTextNoSelect}>$</span>
        <span className={styles.installTextCommand}>curl -o- </span>
        {nvmInstallScriptUrl}
        <span className={styles.installTextCommand}> | bash </span>
      </ShellBox>
      <ShellBox textToCopy="nvm install --lts">
        <span className={styles.installTextNoSelect}>$</span>
        <span className={styles.installTextCommand}>nvm </span>install --lts
      </ShellBox>
      <br />
      <br />
      <LocalizedLink
        className={styles.installDocsButton}
        href="/download/package-manager/#nvm"
      >
        <FormattedMessage id="components.home.installTabs.readDocs" />
      </LocalizedLink>
    </div>
  );
};

const MacOSPanel: FC = () => {
  // @TODO: use hook to get latest version
  const nvmVersion = 'v0.38.0';

  return <PureMacOSPanel nvmVersion={nvmVersion} />;
};

export default MacOSPanel;
