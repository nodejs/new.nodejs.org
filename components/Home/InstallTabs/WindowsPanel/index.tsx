import { FormattedMessage } from 'react-intl';
import LocalizedLink from '../../../LocalizedLink';
import { ShellBox } from '../../../Common';
import styles from '../index.module.scss';
import type { FC } from 'react';

const WindowsPanel: FC = () => (
  <>
    <ShellBox textToCopy="choco install nvs">
      <span className={styles.installTextNoSelect}>$</span>
      <span className={styles.installTextCommand}>choco </span>install nvs
    </ShellBox>
    <ShellBox textToCopy="nvs add lts">
      <span className={styles.installTextNoSelect}>$</span>
      <span className={styles.installTextCommand}>nvs </span>add lts
    </ShellBox>
    <ShellBox textToCopy="nvs use lts">
      <span className={styles.installTextNoSelect}>$</span>
      <span className={styles.installTextCommand}>nvs </span>use lts
    </ShellBox>
    {/* @TODO: Verify this link when we implement content */}
    <LocalizedLink
      className={styles.installDocsButton}
      href="/download/package-manager/#nvs"
    >
      <FormattedMessage id="components.home.installTabs.readDocs" />
    </LocalizedLink>
  </>
);

export default WindowsPanel;
