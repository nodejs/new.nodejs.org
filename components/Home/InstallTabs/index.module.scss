.install {
  background: var(--black10);
  border-radius: var(--space-04);
  box-shadow: 5px 10px 50px rgba(0, 0, 0, 0.25);
  color: var(--black0);
  height: 450px;
  margin: 0 auto;
  margin-bottom: 240px;
  max-width: 832px;
  position: relative;
  width: 90vw;
  z-index: 2;
  .reactTabs {
    -webkit-tap-highlight-color: transparent;

    &TabList {
      display: flex;
      margin: 0;
      padding: 0;
      text-align: center;
    }

    &Tab {
      background: var(--black11);
      border: 1px solid transparent;
      border-bottom: none;
      bottom: -1px;
      color: var(--black4);
      cursor: pointer;
      display: inline-block;
      list-style: none;
      padding: 10px 12px;
      position: relative;
      text-align: center;
      width: 100%;

      &Selected {
        background: var(--black10);
        border-radius: 5px 5px 0 0;
        color: #fff;
      }

      &:focus {
        border-bottom: none;
        border-color: var(--pink5);
        box-shadow: 0 0 5px var(--pink5);
        outline: none;

        &:after {
          background: var(--black10);
          bottom: -5px;
          content: '';
          height: 5px;
          left: 0;
          position: absolute;
          right: 0;
        }
      }
    }

    &TabPanel {
      display: none;
      padding: 20px;

      &Selected {
        display: block;
      }
    }
  }

  &Header {
    font-weight: var(--font-weight-semibold);
    height: 30px;
    margin: 0 5px;

    &Circles {
      float: left;
      height: 100%;

      &Red,
      &Yellow,
      &Green {
        background: var(--black8);
        border-radius: 50%;
        display: inline-block;
        height: 10px;
        margin: 10px 5px;
        width: 10px;
      }

      &Red:hover {
        background: var(--danger5);
      }
      &Yellow:hover {
        background: var(--warning3);
      }
      &Green:hover {
        background: var(--brand5);
      }
    }

    &Text {
      line-height: 30px;
      text-align: center;
      width: calc(100% - 60px);
    }
  }

  &Text {
    color: var(--pink5);

    &Line {
      margin: var(--space-12) 0;
    }

    &NoSelect {
      margin-right: var(--space-12);
      user-select: none;
    }

    &Command {
      color: var(--black0);
    }
  }

  &DocsButton,
  &DocsButton:hover {
    border: 1px solid var(--black3);
    border-radius: var(--space-04);
    bottom: 2rem;
    box-sizing: border-box;
    color: var(--black3);
    font-weight: var(--font-weight-semibold);
    height: 4rem;
    padding: var(--space-08) var(--space-16);
    position: absolute;
    right: 2rem;
    text-decoration: none;
  }
}
