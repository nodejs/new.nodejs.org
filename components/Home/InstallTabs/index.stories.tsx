import InstallTabs from '.';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof InstallTabs>;
type Meta = MetaObj<typeof InstallTabs>;

export const Default: Story = {};

export default { component: InstallTabs } as Meta;
