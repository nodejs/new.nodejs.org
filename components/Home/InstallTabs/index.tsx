import { useEffect, useState } from 'react';
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
import WindowsPanel from './WindowsPanel';
import MacOSPanel from './MacOSPanel';
import LinuxPanel from './LinuxPanel';
import styles from './index.module.scss';
import { detectOS } from '../../../util/detectOS';
import type { UserOS } from '../../../types';
import type { FC, ReactNode, ReactElement } from 'react';

const tabPanelStyles = {
  className: styles.reactTabsTabPanel,
  selectedClassName: styles.reactTabsTabPanelSelected,
};

const getOSPanel: FC<UserOS> = userOS => {
  const createPanel = (...components: ReactNode[]) => {
  return components.map((comp) => (
    <TabPanel {...tabPanelStyles}>{comp}</TabPanel>
  ));
};

switch (userOS) {
  case "MAC":
    return createPanel(<MacOSPanel />, <WindowsPanel />, <LinuxPanel />);
  case "LINUX":
    return createPanel(<LinuxPanel />, <MacOSPanel />, <WindowsPanel />);
  default:
    return createPanel(<WindowsPanel />, <MacOSPanel />, <LinuxPanel />);
}

};

const os = {
  win: 'Windows (Chocolatey)',
  mac: 'macOS (nvm)',
  linux: 'Linux (nvm)',
};

const installTabSystems: Record<UserOS, string[]> = {
  WIN: [os.win, os.mac, os.linux],
  MAC: [os.mac, os.win, os.linux],
  LINUX: [os.linux, os.mac, os.win],
  OTHER: [os.win, os.mac, os.linux],
};

const InstallTabs: FC = () => {
  const [reactTabs, setReactTabs] = useState<ReactNode>();

  useEffect(() => {
    const userOS = detectOS();
    const panelSwitch = getOSPanel(userOS);
    const tabLayout = installTabSystems[userOS];

    setReactTabs(
      <Tabs className={styles.reactTabs}>
        <div className={styles.installHeader}>
          <div className={styles.installHeaderCircles}>
            <div className={styles.installHeaderCirclesRed} />
            <div className={styles.installHeaderCirclesYellow} />
            <div className={styles.installHeaderCirclesGreen} />
          </div>
          <div className={styles.installHeaderText}>
            {userOS === 'MAC' ? 'zsh' : 'bash'}
          </div>
        </div>
        <TabList className={styles.reactTabsTabList}>
          {tabLayout.map(system => (
            <Tab
              className={styles.reactTabsTab}
              selectedClassName={styles.reactTabsTabSelected}
              key={system}
            >
              {system}
            </Tab>
          ))}
        </TabList>
        {tabLayout && panelSwitch}
      </Tabs>
    );
  }, []);

  return <div className={styles.install}>{reactTabs}</div>;
};

export default InstallTabs;
