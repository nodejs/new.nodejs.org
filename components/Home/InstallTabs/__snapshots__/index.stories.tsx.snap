// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home/InstallTabs Default smoke-test 1`] = `
<div class="InstallTabs_install__AmVIK">
  <div class="InstallTabs_reactTabs__bJat9"
       data-rttabs="true"
  >
    <div class="InstallTabs_installHeader__dlshy">
      <div class="InstallTabs_installHeaderCircles__Ib171">
        <div class="InstallTabs_installHeaderCirclesRed___evUB">
        </div>
        <div class="InstallTabs_installHeaderCirclesYellow__HfGea">
        </div>
        <div class="InstallTabs_installHeaderCirclesGreen__9I2jB">
        </div>
      </div>
      <div class="InstallTabs_installHeaderText__nl2HE">
        zsh
      </div>
    </div>
    <ul class="InstallTabs_reactTabsTabList__44kO4"
        role="tablist"
    >
      <li class="InstallTabs_reactTabsTab__bpQEH InstallTabs_reactTabsTabSelected__w3_IO"
          role="tab"
          id="tab:r0:0"
          aria-selected="true"
          aria-disabled="false"
          aria-controls="panel:r0:0"
          tabindex="0"
          data-rttab="true"
      >
        macOS (nvm)
      </li>
      <li class="InstallTabs_reactTabsTab__bpQEH"
          role="tab"
          id="tab:r0:1"
          aria-selected="false"
          aria-disabled="false"
          aria-controls="panel:r0:1"
          data-rttab="true"
      >
        Windows (Chocolatey)
      </li>
      <li class="InstallTabs_reactTabsTab__bpQEH"
          role="tab"
          id="tab:r0:2"
          aria-selected="false"
          aria-disabled="false"
          aria-controls="panel:r0:2"
          data-rttab="true"
      >
        Linux (nvm)
      </li>
    </ul>
    <div class="InstallTabs_reactTabsTabPanel__Hq7es InstallTabs_reactTabsTabPanelSelected__zS6go"
         role="tabpanel"
         id="panel:r0:0"
         aria-labelledby="tab:r0:0"
    >
      <div>
        <pre class="ShellBox_shellBox__n5v7E">
          <div class="ShellBox_top__SxRc6">
            <span>
              SHELL
            </span>
            <button type="button">
              copy
            </button>
          </div>
          <code>
            <span class="InstallTabs_installTextNoSelect__SWhsj">
              $
            </span>
            <span class="InstallTabs_installTextCommand__OPlTv">
              curl -o-
            </span>
            https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh
            <span class="InstallTabs_installTextCommand__OPlTv">
              | bash
            </span>
          </code>
        </pre>
        <pre class="ShellBox_shellBox__n5v7E">
          <div class="ShellBox_top__SxRc6">
            <span>
              SHELL
            </span>
            <button type="button">
              copy
            </button>
          </div>
          <code>
            <span class="InstallTabs_installTextNoSelect__SWhsj">
              $
            </span>
            <span class="InstallTabs_installTextCommand__OPlTv">
              nvm
            </span>
            install --lts
          </code>
        </pre>
        <br>
        <br>
        <a class="InstallTabs_installDocsButton__upRsr"
           href="/en/download/package-manager#nvm"
        >
          Read documentation
        </a>
      </div>
    </div>
    <div class="InstallTabs_reactTabsTabPanel__Hq7es"
         role="tabpanel"
         id="panel:r0:1"
         aria-labelledby="tab:r0:1"
    >
    </div>
    <div class="InstallTabs_reactTabsTabPanel__Hq7es"
         role="tabpanel"
         id="panel:r0:2"
         aria-labelledby="tab:r0:2"
    >
    </div>
  </div>
</div>
`;
